server/mlabns/handlers/lookup.py

```diff
@@ -1,20 +1,68 @@
-from google.appengine.api import memcache
-from google.appengine.api import taskqueue
-from google.appengine.ext import db
-from google.appengine.ext import deferred
-from google.appengine.ext import webapp
-from google.appengine.ext.webapp import template
+import json
+import logging
+import time
 
 from mlabns.db import model
-from mlabns.util import constants
-from mlabns.util import message
 from mlabns.util import lookup_query
+from mlabns.util import message
 from mlabns.util import resolver
 from mlabns.util import util
 
-import json
-import logging
-import time
+from google.appengine.ext import webapp
+from google.appengine.ext.webapp import template
+
+
+def _create_af_aware_fqdn(fqdn, address_family):
+    """Adds the v4/v6 only annotation to the fqdn.
+
+    Example:
+        fqdn:       'npad.iupui.mlab3.ath01.measurement-lab.org'
+        ipv4 only:  'npad.iupui.mlab3v4.ath01.measurement-lab.org'
+        ipv6 only:  'npad.iupui.mlab3v6.ath01.measurement-lab.org'
+
+    Args:
+        fqdn: A tool FQDN with no address family specific annotation.
+        address_family: The address family for which to create the FQDN or None
+            to create an address family agnostic FQDN.
+
+    Returns:
+        A FQDN specific to a particular address family, or the original FQDN
+        if an address family is not specified.
+    """
+    if address_family == message.ADDRESS_FAMILY_IPv4:
+        fqdn_annotation = 'v4'
+    elif address_family == message.ADDRESS_FAMILY_IPv6:
+        fqdn_annotation = 'v6'
+    elif not address_family:
+        fqdn_annotation = ''
+    else:
+        logging.error('Unrecognized address family: %s', address_family)
+        return fqdn
+
+    fqdn_parts = fqdn.split('.')
+    fqdn_parts[2] += fqdn_annotation
+
+    return '.'.join(fqdn_parts)
+
+
+def _create_tool_url(fqdn, address_family, http_port):
+    """Create an HTTP URL for a tool.
+
+    Args:
+        fqdn: A tool FQDN with no address family specific annotation.
+        address_family: The address family for which to create the URL or None
+            to create an address family agnostic URL.
+        http_port: The HTTP port number part of the URL.
+
+    Returns:
+        A tool URL, annotated with the appropriate address family (when
+        available).
+    """
+    url = 'http://' + _create_af_aware_fqdn(fqdn, address_family)
+    if http_port:
```

Comment:
Good point. It's actually a check for null. This made me realize that it's actually illegal to call this function if `http_port` is not set because we shouldn't be constructing HTTP URLs for slices that have no HTTP port assigned (i.e. can't speak HTTP), so I got rid of the `if` in `_create_tool_url` entirely.

Fixed in b20a39ee1f4c57b5091ec350767d1430524e7d7a